pkg/config/config_template.yaml

```diff
@@ -1,857 +1,1235 @@
+#   <HIGH_CARDINALITY_ANNOTATION>: +<TAG_KEY>
+
 {{ end -}}
 {{- if .ECS }}
-# ECS integration
+
+###################################
+## ECS integration Configuration ##
+###################################
+
+## @param ecs_agent_container_name - string - optional - default: ecs-agent
+## The ECS Agent container should be autodetected when running with the
+## default (ecs-agent) name. If not, change the container name here:
 #
-# The ECS agent container should be autodetected when running with the
-# default (ecs-agent) name. Else, you can change the container name the
-# agent will look for, or force a fixed url:
 # ecs_agent_container_name: ecs-agent
-# ecs_agent_url: http://localhost:51678
-#
-# Fargate clusters use other endpoints and are not affected by these options.
+
+## @param ecs_agent_url - string - optional - default: http://localhost:51678
+## The ECS Agent container should be autodetected when running with the
+## default (ecs-agent) name. If not, change the container name the
+## Agent should look for with ecs_agent_container_name, or force a fixed url here:
 #
+# ecs_agent_url: http://localhost:51678
+
 {{ end -}}
 {{- if .CRI }}
-# CRI integration
+
+###################################
+## CRI integration Configuration ##
+###################################
+
+## @param cri_socket_path - string - optional - default: ""
+## To activate the CRI check, indicate the path of the CRI socket you're using
+## and mount it in the container if needed.
+## If left empty, the CRI check is disabled.
+## see: https://docs.datadoghq.com/integrations/cri/
 #
-# To activate the CRI check you'll need to indicate the path of the
-# CRI runtime you're using (and mount it in the container if needed)
-# cri_socket_path: /var/run/containerd/containerd.sock
+# cri_socket_path: ""
+
+## @param cri_connection_timeout - integer - optional - default: 5
+## Configure the initial connection timeout in seconds.
 #
-# You can configure the initial connection timeout (in seconds)
-# cri_connection_timeout: 1
+# cri_connection_timeout: 5
+
+## @param cri_query_timeout - integer - optional - default: 5
+## Configure the timeout in seconds for querying the CRI.
 #
-# You can configure the timeout (in seconds) for querying the CRI
 # cri_query_timeout: 5
-#
+
 {{ end -}}
 {{- if .Containerd}}
-# Containerd integration
+
+##########################################
+## Containerd integration Configuration ##
+##########################################
+
+## @param cri_socket_path - string - optional - default: /var/run/containerd/containerd.sock
+## To activate the Containerd check, indicate the path of the Containerd socket you're using
+## and mount it in the container if needed.
+## see: https://docs.datadoghq.com/integrations/containerd/
 #
-# To activate the Containerd check you'll need to indicate the path of the
-# Containerd socket you're using (and mount it in the container if needed)
 # cri_socket_path: /var/run/containerd/containerd.sock
```

Comment:
Isn't it weird to have the same config parameter `cri_socket_path` & `cri_query_timeout` twice?

`Containerd` & `CRI` conditions are hardcoded to `true`

Reply:
that's indeed not useful, let's merge this PR and address this in another PR